### ISSUE TITLE
HaplotypeCaller 4.1.2.0 crashes in some locations when running in GENOTYPE_GIVEN_ALLELES mode

### ISSUE THREAD

**Issue by TedBrookings** [broadinstitute/gatk#6061]
### Affected tool(s) or class(es)
_HaplotypeCaller_

### Affected version(s)
- 4.1.2.0  (4.1.1.0 does *NOT* crash, nor does latest master, as of commit 4798f6b7c02959ef4101bedd7eff8283f94423b2)

### Description 
HaplotypeCaller crashes with the following stack trace:

> java.lang.IllegalArgumentException: Cigar cannot be null
>         at org.broadinstitute.hellbender.utils.read.AlignmentUtils.consolidateCigar(AlignmentUtils.java:716)
>         at org.broadinstitute.hellbender.utils.haplotype.Haplotype.setCigar(Haplotype.java:193)
>         at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.AssemblyBasedCallerUtils.addGivenAlleles(AssemblyBasedCallerUtils.java:350)
>         at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.AssemblyBasedCallerUtils.assembleReads(AssemblyBasedCallerUtils.java:291)
>         at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCallerEngine.callRegion(HaplotypeCallerEngine.java:542)
>         at org.broadinstitute.hellbender.tools.walkers.haplotypecaller.HaplotypeCaller.apply(HaplotypeCaller.java:240)
>         at org.broadinstitute.hellbender.engine.AssemblyRegionWalker.processReadShard(AssemblyRegionWalker.java:308)
>         at org.broadinstitute.hellbender.engine.AssemblyRegionWalker.traverse(AssemblyRegionWalker.java:281)
>         at org.broadinstitute.hellbender.engine.GATKTool.doWork(GATKTool.java:1039)
>         at org.broadinstitute.hellbender.cmdline.CommandLineProgram.runTool(CommandLineProgram.java:139)
>         at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMainPostParseArgs(CommandLineProgram.java:191)
>         at org.broadinstitute.hellbender.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:210)
>         at org.broadinstitute.hellbender.Main.runCommandLineProgram(Main.java:162)
>         at org.broadinstitute.hellbender.Main.mainEntry(Main.java:205)
>         at org.broadinstitute.hellbender.Main.main(Main.java:291)
> 

#### Steps to reproduce
This is the invoking command. Critical features are the particular region (only some regions crash), GENOTYPE_GIVEN_ALLELES, and assembly-region-padding.

```
gatk HaplotypeCaller --java-options '-XX:GCTimeLimit=50 -XX:GCHeapFreeLimit=10 -Xmx186260m' --spark-runner LOCAL \
    -R Homo_sapiens_assembly38.fasta \
    -I NA12878_2x_chr16_76282200-76284450.bam \
    -O NA12878_2x_chr16_76282200-76284450.vcf.gz \
    --genotyping-mode GENOTYPE_GIVEN_ALLELES \
    --use-alleles-trigger true \
    --alleles 00-common-all.hs38DH_chr16_76282200-76284450.vcf.gz \
    --dbsnp 00-common-all.hs38DH_chr16_76282200-76284450.vcf.gz \
    -L chr16:76282200-76284450 \
    --assembly-region-padding 100

```
I've placed versions of the needed here (chopped to small size so its easier to move them around):
gs://broad-dsde-methods-tbrookin/HaplotypeCallerCrash/

#### Expected behavior
_HaplotypeCaller should run to completion_

#### Actual behavior
_HaplotypeCaller crashes_

**Comment by cwhelan** [comment]
I think

**Comment by cwhelan** [comment]
Sorry, did not mean to close, just clicked the wrong button. Will let @davidbenjamin close by linking to his PR.

**Comment by davidbenjamin** [comment]
Closed by PR #6047.
